repository.go

```diff
@@ -123,6 +148,43 @@ func (r *mongoRepository) queryUser(ctx context.Context, filter interface{}, opt
 	}, nil
 }
 
+func (r *mongoRepository) queryUser(ctx context.Context, filter interface{}, opts ...*options.FindOneOptions) (user, error) {
+	users := r.users()
+
+	res := users.FindOne(ctx, filter, opts...)
+	if res.Err() != nil && errors.Is(mongo.ErrNoDocuments, res.Err()) {
```

Comment:
The correct syntax is `errors.Is(actualError, targetError)`, so this has to be `errors.Is(res.Err(), mongo.ErrNoDocuments)`.